### PR TITLE
Bugfix: Fixed weapon FOV issues

### PR DESCRIPTION
**Pull request by TW1STaL1CKY**
1. Fixed player's FOV being stuck zoomed in if their weapon is removed while they're scoped in
2. Fixed FOV transitions being delayed in DynamicCamera when the player has lag, eg. when unscoping with a rifle

### CODE REVIEW

File: gamemodes/terrortown/gamemode/client/cl_main.lua [hunk 1]
```diff
@@ -867,12 +869,25 @@ function GM:DynamicCamera(viewTable, ply)
...
-            (CurTime() - ply:GetFOVTime()) * game.GetTimeScale() * cvHostTimescale:GetFloat()
-        )
+        local fovTime = ply:GetFOVTime()
+
+        if lastFovTimeValue ~= fovTime then
+            lastFovTimeValue = fovTime
+            curTimeShift = 0
+        end
+
+        local curTime, fovTranDur = CurTime() + curTimeShift, ply:GetFOVTransitionTime()
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/client/cl_main.lua:879]:
```suggestion
        local curTime = CurTime() + curTimeShift
        local fovTransitionDuration = ply:GetFOVTransitionTime()
```

**Review by TimGoll — CHANGES_REQUESTED** [review]
Looks almost good to me. Please make sure to not use abbreviations in variable names if possible. And there is a style error left.

Thanks for the fix!

File: gamemodes/terrortown/gamemode/client/cl_main.lua (hunk 1 above)
Comment (TW1STaL1CKY) [reply gamemodes/terrortown/gamemode/client/cl_main.lua]: Done
I also realised I didn't need fovTransitionDuration in a local anymore

**Review by TimGoll — APPROVED** [review]

**Review by saibotk — APPROVED** [review]
Thank you very much for the PR 🩷

**Pull request merged**